Feat/#1198 knowledge base change order

AndrewDavydenko opened the pull request:
-   [x] Have you followed the [contributing guidelines](https://github.com/ever-co/gauzy/blob/master/.github/CONTRIBUTING.md)?
-   [x] Have you explained what your changes do, and why they add value?

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

---
added custom drag&drop (now you can't move a category to the article) 
added deleting node from the tree when you move it
added new data for the tree
add to the new place - in progress
add the tree entity&recursive seeding - in progress

AndrewDavydenko commented:
web and API worked, but after pulling got build problems

AndrewDavydenko commented:
added more icons for choosing
added language and color choosing (for lang, in my opinion, we don't need it because we have global lang choosing)
UI fix with adding icon
in progress working with the tree in the API(asked @rmagon in slack for that issue)

rmagon commented:
Hi @AndrewDavydenko ... I took a look at your PR, I think you may have got the relations upside down. 

One entity will have many children and many entities will have one parent

Your entity should look like this:
```
@ManyToOne((type) => HelpCenter, (children) => children.children, {
 cascade: ["insert"],
 nullable: true
})
parent?: IHelpCenter;

@OneToMany((type) => HelpCenter, (children) => children.parent, {
  cascade: ["insert"]
})
children?: IHelpCenter[];
```

Also, your seed file save will then look like this:
```
  for(const h of helpCenterMenuList) {
    const entity = await createEntity(connection, h);
    await save(connection, entity);
  }
```

where save and createEntity are:
```
const save = async (
	connection: Connection,
	node: IHelpCenter
): Promise<void> => {
	await connection.manager.save(node);
};


const createEntity = async (connection: Connection, node: IHelpCenter) => {
  if(!node) {
    return;
  }
  return connection.manager.create(HelpCenter, node);

}
```

And after seed your table will look like:
<img width="1403" alt="Screenshot 2020-06-06 at 1 41 00 AM" src="https://user-images.githubusercontent.com/6750734/83918647-cc22ff00-a796-11ea-9ad3-56e654908946.png">

AndrewDavydenko commented:
https://www.loom.com/share/f7434b6194ed43e3bd4216b5bb65c427
added seeding with recursive
added drag&drop to the database
in progress sorting by index
in progress toastrs
in progress bugfix with safeHtml
